src/_guides/environment-declarations.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Configuring apps with environment declarations
+description: >
+  Learn about using environment declarations to customize application behavior.
+---
+
+Dart supports specifying environment declarations when
+building or running an application.
+Environment declarations are used to specify
+configuration values or options as key-value pairs
+that are accessed and evaluated at compile time.
+
+Your app can use the values of environment declarations
+to change its functionality or behavior.
+Dart compilers can take advantage of this to eliminate code
+no longer reached due to the values specified through tree shaking.
+
+Some potential use cases for defining environment declarations are:
+
+* Adding functionality during debugging, such as enabling logging.
+* Creating separate flavors of your application.
+* Configuring application behavior, such as the port of an HTTP server.
+* Enabling an experimental mode of your application for testing.
+* Switching between testing and production backends.
+
+To specify an environment declaration
+to an application with the [`dart run`][] or [`dart compile`][] subcommands,
+use the `-D` or `--define` options and specify `<NAME>=<VALUE>`:
+
+```terminal
+$ dart run --define=DEBUG=true -DFLAVOR=free
+```
+
+To learn more about specifying environment declarations,
+including for other tools and compilers, see
+[Specifying environment declarations][].
+
+[`dart run`]: /tools/dart-run
+[`dart compile`]: /tools/dart-compile
+[Specifying environment declarations]: #specifying-environment-declarations
+
+## Accessing environment declarations
+
+To access specified environment declaration values,
+use one of the `fromEnvironment` constructors
+with `const` or within a constant context.
+Use [`bool.fromEnvironment`][bool-from] for `true` or `false` values,
+[`int.fromEnvironment`][int-from] for integer values,
+and [`String.fromEnvironment`][string-from] for anything else.
+
+{{site.alert.note}}
+  The environment declaration constructors are only guaranteed
+  to work when invoked as `const`.
+  Most compilers must be able to evaluate their value at compile time.
+{{site.alert.end}}
+
+Each of the `fromEnvironment` constructors require the
+name or key of the environment declaration key-value pair,
+as well as an optional `defaultValue` override
+that is used when a declaration wasn't defined.
```

Comment:
, or the value string wasn't valid as an `int` or `bool` result.

(The default value is used for `int.fromEnv...` when the value string cannot be parsed as an integer, and for `bool.fromEnv...` when the value isn't `true` or `false`.)